CMakeLists.txt

```
@@ -39,172 +97,66 @@ option(USE_ATOMIC_COLLISION_WRITE
-else()
-endif()
-
-# Make better loading Linux binaries
-set(CMAKE_BUILD_WITH_INSTALL_RPATH ON)
-set(CMAKE_INSTALL_RPATH "$ORIGIN")
-
-# Detect library version
-file(READ "src/native/NativeConstants.cs" versionFile)
-
-string(REGEX MATCH "Version = ([0-9]+);" _ "${versionFile}")
-set(NATIVE_LIBRARY_VERSION ${CMAKE_MATCH_1})
-
-if(NOT NATIVE_LIBRARY_VERSION)
-  message(FATAL_ERROR "Failed to parse native library version")
-endif()
-
-string(REGEX MATCH "EarlyCheck = ([0-9]+);" _ "${versionFile}")
-set(EARLY_CHECK_LIBRARY_VERSION ${CMAKE_MATCH_1})
-
-if(NOT EARLY_CHECK_LIBRARY_VERSION)
-  message(FATAL_ERROR "Failed to parse native (early check) library version")
-endif()
-
-string(REGEX MATCH "ExtensionVersion = ([0-9]+);" _ "${versionFile}")
-set(THRIVE_EXTENSION_VERSION ${CMAKE_MATCH_1})
-
-if(NOT THRIVE_EXTENSION_VERSION)
-  message(FATAL_ERROR "Failed to parse Thrive GDExtensions library version")
-endif()
+option(THRIVE_GODOT_API_FILE "Override folder Godot API file is looked for in"
+ "")
 
 message(STATUS "Configured native library version ${NATIVE_LIBRARY_VERSION}")
 
 # Configure include file
-configure_file("src/native/Include.h.in" "${PROJECT_BINARY_DIR}/Include.h")
-include_directories(${PROJECT_BINARY_DIR})
+configure_file(
+    "${CMAKE_CURRENT_SOURCE_DIR}/src/native/Include.h.in"
+    "${CMAKE_BINARY_DIR}/Include.h"
+    @ONLY
+)
+
+include_directories(${CMAKE_BINARY_DIR})
 
 # Configure gdextension stuff
 if(THRIVE_GODOT_API_FILE)
-  set(GODOT_GDEXTENSION_DIR "${THRIVE_GODOT_API_FILE}")
-  include_directories("${THRIVE_GODOT_API_FILE}")
-  message(STATUS "Using custom location of Godot API file")
+    set(GODOT_GDEXTENSION_DIR "${THRIVE_GODOT_API_FILE}")
 else()
-  set(GODOT_GDEXTENSION_DIR "${CMAKE_BINARY_DIR}/api")
-  include_directories("${CMAKE_BINARY_DIR}/api")
+    set(GODOT_GDEXTENSION_DIR "${CMAKE_BINARY_DIR}/api")
 endif()
 
-# Float precision configuration for Godot
-set(FLOAT_PRECISION "single")
+include_directories(${GODOT_GDEXTENSION_DIR})
 
 # GODOT_CPP_SYSTEM_HEADERS
 # GODOT_CPP_WARNING_AS_ERROR
 
 # Add the subfolders that define the actual things to build
 add_subdirectory(third_party)
 
+# Thrive AVX check location has been called only here 
 if(NOT THRIVE_DISTRIBUTION OR (THRIVE_AVX AND THRIVE_DISTRIBUTION))
-  # Early checks is not compiled in non-avx mode
-  add_subdirectory(src/native/early_checks)
+    add_subdirectory(src/native/early_checks)
 endif()
 
 add_subdirectory(src/native)
 add_subdirectory(src/extension)
 
+install(TARGETS thrive_native thrive_extension
+    RUNTIME DESTINATION bin
+    LIBRARY DESTINATION lib
+    ARCHIVE DESTINATION lib
+)
+
+if(MSVC AND CMAKE_BUILD_TYPE STREQUAL "Debug")
+    install(FILES 
+        $<TARGET_PDB_FILE:thrive_native>
+        $<TARGET_PDB_FILE:thrive_extension>
+        DESTINATION bin 
+        OPTIONAL
+    )
+endif()
+
+message(STATUS "Building Thrive version ${NATIVE_LIBRARY_VERSION}")
```

```suggestion
message(STATUS "Building ThriveNative version ${NATIVE_LIBRARY_VERSION}")
```

also pretty weird that only one library's version is mentioned.